### PR TITLE
Add tests to ensure global variables are not prohibited

### LINKED ISSUE #3: Support magic numbers when used to set a GLOBAL variable
## What 

Support the use of magic numbers when setting a global variable 

``` ruby 
# No offence 
$MY_GLOBAL_VAR = 1

# No offence 
def my_method 
  $MY_GLOBAL_VAR = 1
end
```

## Why 

This pattern is typically used as part of global configuration (hopefully very sparingly), which can often involve using integer and float values. Variable names should be descriptive enough

### PR DESCRIPTION
**Pull request by Bodacious**
## What 

Adds tests to ensure global variables can be assigned to magic numbers

## Why 

When adding rules to detect other types of magic numbers, we want to ensure these aren't reported as false positives 

Close #3

### CODE REVIEW

File: test/test_helper.rb [hunk 1]
```diff
@@ -10,9 +10,11 @@ def assert_offense(violation_message)
     assert_equal(cop.offenses.first.message, violation_message) if cop.offenses.any?
   end
 
-  def refute_offense
-    assert_empty(cop.offenses)
+  def assert_no_offenses(cop_name = nil)
+    matching_offenses = cop_name.nil? ? cop.offenses : cop.offenses.select { _1.cop_name == cop_name }
+    assert_empty(matching_offenses, "Expected no offense to be detected but there was one")
   end
+  alias refute_offense assert_no_offenses
```
Comment (Bodacious) [inline test/test_helper.rb:17]: We added these on two parallel branches. Have merged in my method definition and aliased to the name used in your specs

File: test/lib/no_magic_numbers_test.rb [hunk 2]
```diff
@@ -103,7 +127,7 @@ def test_method
         end
       RUBY
 
-      refute_offense
+      assert_no_offenses("Custom/NoMagicNumbers")
```
Comment (Bodacious) [inline test/lib/no_magic_numbers_test.rb]:
```suggestion
      refute_offense
```

File: test/lib/no_magic_numbers_test.rb [hunk 3]
```diff
@@ -86,14 +86,38 @@ def test_method
...
+
     def test_ignores_magic_integers_assigned_via_class_writers_on_another_object
       inspect_source(<<~RUBY)
         def test_method
           Foo.klass_method = 1
         end
       RUBY
 
-      refute_offense
+      assert_no_offenses("Custom/NoMagicNumbers")
```
Comment (Bodacious) [inline test/lib/no_magic_numbers_test.rb]:
```suggestion
      refute_offense
```

File: test/test_helper.rb (hunk 1 above)
Comment (AgentAntelope) [reply test/test_helper.rb:17]: Sounds good! Might want to rebase and remove my definition, I can't recall what alias does with a clashing method

**Review by AgentAntelope — APPROVED** [review]

**Pull request merged**